### PR TITLE
add LoadingFutureBuilder

### LINKED ISSUE #249: Loading Future builder
Future builder with onData, loading and error widgets

### PR DESCRIPTION
**Pull request by DasProffi**
closes #249

### CODE REVIEW

File: packages/helpwave_widget/lib/src/loading/load_error_widget.dart [hunk 1]
```diff
@@ -32,7 +32,7 @@ class LoadErrorWidget extends StatelessWidget {
             color: iconColor,
           ),
           const SizedBox(height: distanceBig),
-          Text(errorText),
+          Text(errorText ?? "Error on Load"),
```
Comment (use-to) [inline packages/helpwave_widget/lib/src/loading/load_error_widget.dart:35]: We should provide translations for such general error messages.

File: packages/helpwave_widget/lib/src/loading/loading_spinner.dart [hunk 2]
```diff
@@ -34,12 +34,12 @@ class LoadingSpinner extends StatelessWidget {
...
             child: CircularProgressIndicator(
               strokeWidth: width,
-              semanticsLabel: text,
+              semanticsLabel: text ?? "Loading",
               color: color,
             ),
           ),
           const SizedBox(height: distanceBig),
-          Text(text),
+          Text(text ?? "Loading"),
```
Comment (use-to) [inline packages/helpwave_widget/lib/src/loading/loading_spinner.dart:42]: Provide a translation

File: packages/helpwave_widget/lib/src/loading/loading_spinner.dart [hunk 3]
```diff
@@ -34,12 +34,12 @@ class LoadingSpinner extends StatelessWidget {
             width: size,
             child: CircularProgressIndicator(
               strokeWidth: width,
-              semanticsLabel: text,
+              semanticsLabel: text ?? "Loading",
```
Comment (use-to) [inline packages/helpwave_widget/lib/src/loading/loading_spinner.dart:37]: See below

**Review by use-to — CHANGES_REQUESTED** [review]

File: packages/helpwave_widget/lib/src/loading/load_error_widget.dart (hunk 1 above)
Comment (DasProffi) [reply packages/helpwave_widget/lib/src/loading/load_error_widget.dart:35]: Currently, the localization package isn't required in the widget package, therefore we have no access to it, do we want to change this?

Comment (use-to) [reply packages/helpwave_widget/lib/src/loading/load_error_widget.dart:35]: Let's discuss that tomorrow, okay?

**Comment by florian-sabonchi** [comment]
What do you think about mark this pull request as draft if it is not actively developed?

**Comment by DasProffi** [comment]
The pr is done, since all conversation are closed and only require a review by @use-to therefore the current state of the pr is correct

**Review by use-to — APPROVED** [review]

**Comment by use-to** [comment]
Allright, yeah I'm sorry! Very busy atm!

**Pull request merged**